query/src/test/java/org/infinispan/query/queries/reindex/ReindexTest.java

```diff
@@ -0,0 +1,77 @@
+package org.infinispan.query.queries.reindex;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.lucene.search.Query;
+import org.infinispan.Cache;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.cache.Index;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.query.CacheQuery;
+import org.infinispan.query.Search;
+import org.infinispan.query.SearchManager;
+import org.infinispan.query.test.ReindexTestObject;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+/**
+ * Created by Seto on 2017/3/15.
+ */
+@Test(groups = {"functional", "smoke"}, testName = "query.queries.reindex.ReindexTest")
+public class ReindexTest extends SingleCacheManagerTest {
```

Comment:
This test looks fine, but it only tests ```LOCAL``` caches and a single ```IndexManager```.
To broaden the coverage, I recommend adding a test method to ```org.infinispan.query.distributed.DistributedMassIndexingTest``` to partially re-index some keys, since this test is inherited to cover different index managers, cache types and configurations.

The way the test work could is simple: we have a flag called ```Flag.SKIP_INDEXING``` to bypass the indexing for a single entry, example:

```cache.getAdvancedCache().withFlags(Flag.SKIP_INDEXING).put(K,V)```

the test could simply put something with this flag, verify the entry is not indexed (zero results in searches) and then call the new method to reindex just that single entry and re-query it.